### PR TITLE
updated model attributes

### PR DESCRIPTION
**Pull request by mohamedshabanofficial**
solves #17288 #17369

### CODE REVIEW

File: sdk/documenttranslation/azure-ai-documenttranslation/azure/ai/documenttranslation/_models.py [hunk 1]
```diff
@@ -233,6 +233,7 @@ class JobStatusResult(object):  # pylint: disable=useless-object-inheritance, to
     :ivar int documents_not_yet_started_count: Number of documents that have not yet started being translated.
     :ivar int documents_cancelled_count: Number of documents that were cancelled for translation.
     :ivar int total_characters_charged: Total characters charged across all documents within the job.
+    :ivar int has_completed: boolean to check whether a document finished translation or not.
```
Comment (kristapratico) [inline sdk/documenttranslation/azure-ai-documenttranslation/azure/ai/documenttranslation/_models.py:236]: This should be `:ivar bool has_completed:`

I think we'll need to be a bit more explicit about what this is in the docstrings. Something like:
"If the status returned indicates that the translation job has completed. A translation job is considered 'complete' if it has reached a terminal state like 'Succeeded', 'Cancelled', or 'Failed'."

edit: same comment for the DocumentStatusResult.has_completed

File: sdk/documenttranslation/azure-ai-documenttranslation/azure/ai/documenttranslation/_models.py [hunk 2]
```diff
@@ -268,13 +270,16 @@ def _from_generated(cls, batch_status_details):
             documents_in_progress_count=batch_status_details.summary.in_progress,
             documents_not_yet_started_count=batch_status_details.summary.not_yet_started,
             documents_cancelled_count=batch_status_details.summary.cancelled,
-            total_characters_charged=batch_status_details.summary.total_character_charged
+            total_characters_charged=batch_status_details.summary.total_character_charged,
+            has_completed=True if batch_status_details.status not in ["NotStarted", "Running"] else False
```
Comment (kristapratico) [inline sdk/documenttranslation/azure-ai-documenttranslation/azure/ai/documenttranslation/_models.py:274]: I think "Cancelling" should also be in this list. edit: same for below

File: sdk/documenttranslation/azure-ai-documenttranslation/azure/ai/documenttranslation/_models.py [hunk 3]
```diff
@@ -233,6 +233,10 @@ class JobStatusResult(object):  # pylint: disable=useless-object-inheritance, to
     :ivar int documents_not_yet_started_count: Number of documents that have not yet started being translated.
     :ivar int documents_cancelled_count: Number of documents that were cancelled for translation.
     :ivar int total_characters_charged: Total characters charged across all documents within the job.
+    :ivar bool has_completed: boolean to check whether a document finished translation or not.
```
Comment (kristapratico) [inline sdk/documenttranslation/azure-ai-documenttranslation/azure/ai/documenttranslation/_models.py:236]: not sure if you meant to keep this first sentence?

File: sdk/documenttranslation/azure-ai-documenttranslation/tests/testcase.py [hunk 4]
```diff
@@ -146,6 +148,10 @@ def _validate_translation_job(self, job_details, **kwargs):
         inprogress = kwargs.pop('inprogress', None)
         notstarted = kwargs.pop('notstarted', None)
         cancelled = kwargs.pop('cancelled', None)
+        
+        has_completed = False
+        if status:
+            has_completed = True if status not in ["NotStarted", "Running"] else False
```
Comment (kristapratico) [inline sdk/documenttranslation/azure-ai-documenttranslation/tests/testcase.py:154]: "Cancelling" here as well

File: sdk/documenttranslation/azure-ai-documenttranslation/azure/ai/documenttranslation/_models.py (hunk 3 above)
Comment (mohamedshabanofficial) [reply sdk/documenttranslation/azure-ai-documenttranslation/azure/ai/documenttranslation/_models.py:236]: i think this explains the attribute in a nutshell
then after that, we have more detailed explanation

Comment (kristapratico) [reply sdk/documenttranslation/azure-ai-documenttranslation/azure/ai/documenttranslation/_models.py:236]: This is for the job, though. Do you mean to say "whether the translation job finished or not"?

Comment (mohamedshabanofficial) [reply sdk/documenttranslation/azure-ai-documenttranslation/azure/ai/documenttranslation/_models.py:236]: ok will remove it

Comment (mohamedshabanofficial) [reply sdk/documenttranslation/azure-ai-documenttranslation/azure/ai/documenttranslation/_models.py:236]: sorry will correct it!

Comment (mohamedshabanofficial) [reply sdk/documenttranslation/azure-ai-documenttranslation/azure/ai/documenttranslation/_models.py]: so! is it ok now?

Comment (kristapratico) [reply sdk/documenttranslation/azure-ai-documenttranslation/azure/ai/documenttranslation/_models.py]: yes, thank you! going to approve and merge!

**Review by kristapratico — APPROVED** [review]

**Comment by mohamedshabanofficial** [comment]
@kristapratico it's ready for merge

**Pull request merged**